packages/firebase_database/firebase_database/lib/src/database_reference.dart

```diff
@@ -165,29 +165,27 @@ class DatabaseReference extends Query {
           map['error'] != null ? DatabaseError._(map['error']) : null;
       final bool committed = map['committed'];
       final DataSnapshot? dataSnapshot = map['snapshot'] != null
-          ? DataSnapshot._fromJson(map['snapshot'], map['childKeys'])
+          ? DataSnapshot._fromJson(map['snapshot'], null)
```

Comment:
Native never sends any `childKeys` parameters. So I just removed it

Reply:
Is this right? cc @russellwheatley since I think you added the `childKeys` thing right so could you confirm? Thanks!

Reply:
I looked at both the java and objective C code and didn't see a `childKeys`

Maybe it wasn't implemented for transactions.

Reply:
It's not used in transactions, but it is used in every event listener with a reference to the database.

[Here](https://github.com/FirebaseExtended/flutterfire/blob/2ca476431a2a8ed262080bf7282915a3b893f429/packages/firebase_database/firebase_database/android/src/main/java/io/flutter/plugins/firebase/database/MethodCallHandlerImpl.java#L211) is where `childKeys` is appended to the map passed back to Flutter.

[Here](https://github.com/FirebaseExtended/flutterfire/blob/2ca476431a2a8ed262080bf7282915a3b893f429/packages/firebase_database/firebase_database/android/src/main/java/io/flutter/plugins/firebase/database/MethodCallHandlerImpl.java#L236-L260) are all the event types that pass back `childKeys`.

[Here](https://github.com/FirebaseExtended/flutterfire/blob/2ca476431a2a8ed262080bf7282915a3b893f429/packages/firebase_database/firebase_database/lib/src/event.dart#L22) is where the event creates a `DataSnapshot` where `childKeys` is expected in the database package.

Reply:
So there are still cases where `runTransaction` can receive a `childKeys`?

Reply:
no, not `runTransaction` from what i can see. It is only for events.